nodes/logic/evolver.py

```diff
@@ -0,0 +1,452 @@
+    )
+
+from mathutils.noise import seed_set, random
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import updateNode
+from sverchok.core.update_system import make_tree_from_nodes, do_update
+
+Gene = namedtuple('Gene', 'name g_type, min_n, max_n, range, init_val')
+
+evolver_mem = {}
+
+def is_valid_node(node, genotype_frame):
+    if genotype_frame == 'All' and node.bl_idname == "SvNumberNode":
+        return True
+    if node.parent and node.parent.name == genotype_frame and node.bl_idname == "SvNumberNode":
+        return True
+    return False
+
+def get_genes(target_tree, genotype_frame):
+    genes = []
+    for node in target_tree.nodes:
+        if is_valid_node(node, genotype_frame):
+            num_type = node.selected_mode
+            if node.selected_mode == "float":
+                min_n = node.float_min
+                max_n = node.float_max
+                initial_value = node.float_
+            else:
+                min_n = node.int_min
+                max_n = node.int_max
+                initial_value = node.int_
+            gene = Gene(name=node.name, g_type=num_type, min_n=min_n, max_n=max_n, range=max_n-min_n, init_val=initial_value)
+            genes.append(gene)
+    return genes
+
+class DNA:
+    genes_def = []
+    genes = []
+    fitness = 0
+    def __init__(self, genes_def, random_val=True, empty=False):
+        self.genes_def = genes_def
+        self.genes = []
+        self.fitness = 0
+        if empty:
+            return
+        if random_val:
+            for gene in genes_def:
+                agent_gene = gene.min_n + random() * gene.range
+                if gene.g_type == 'int':
+                    agent_gene = int(agent_gene)
+                self.genes.append(agent_gene)
+        else:
+            for gene in genes_def:
+                agent_gene = gene.init_val
+                self.genes.append(agent_gene)
+
+    def evaluate_fitness(self, tree, update_list, node):
+        tree.sv_process = False
+        for gen_data, agent_gene in zip(self.genes_def, self.genes):
+            if gen_data.g_type == 'float':
+                tree.nodes[gen_data.name].float_ = agent_gene
+            else:
+                tree.nodes[gen_data.name].int_ = agent_gene
+
+        tree.sv_process = True
+        do_update(update_list, tree.nodes)
+        agent_fitness = node.inputs[0].sv_get(deepcopy=False)[0]
+        if isinstance(agent_fitness, list):
+            agent_fitness = agent_fitness[0]
+        self.fitness = agent_fitness
+        # return agent_fitness
+    def cross_over(self, other_ancestor, mutation):
+        # new_agent = []
+        new_agent = DNA(self.genes_def, empty=True)
+        for g, g1, o_gene in zip(self.genes, other_ancestor.genes, self.genes_def):
+            if random() < mutation:
+                if random() < 0.5:
+                    #total gene reset
+                    new_gene = o_gene.min_n + random() * o_gene.range
```

Comment:
It probably could be written in more human way. I don't know exactly how it should but something in a way like this:
```suggestion
            next_mutation_success= random()
            if next_mutation_success < mutation:
                success_barrier = 0.5
                if next_mutation_success < success_barrier:
                    #total gene reset
                    new_gene = o_gene.min_n + random() * o_gene.range  # I believe this string also can be improved
```